Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIPermissions.mm

```diff
@@ -0,0 +1,204 @@
+        @"host_permissions": @[ @"*://*.apple.com/*" ]
+    };
+
+    auto *backgroundScript = Util::constructScript(@[
+        // contains() should return true for all named permissions and granted match patterns.
+        @"browser.test.assertTrue(await browser.permissions.contains({'permissions': ['alarms', 'activeTab']}))",
+        @"browser.test.assertTrue(await browser.permissions.contains({'origins': ['*://webkit.org/*']}))",
+
+        // contains() should return false for non granted match patterns.
+        @"browser.test.assertFalse(await browser.permissions.contains({'origins': ['*://apple.com/*']}))",
+
+        // Removing optional permissions should pass.
+        @"browser.test.assertTrue(await browser.permissions.remove({'permissions': ['cookies']}))",
+
+        // Removing functional permissions should fail.
+        @"browser.test.assertRejects(browser.permissions.remove({'permissions': ['alarms']}))",
+        @"browser.test.assertRejects(browser.permissions.remove({'permissions': ['alarms', 'activeTab']}))",
+        @"browser.test.assertRejects(browser.permissions.remove({'permissions': ['cookies', 'activeTab']}))",
+        @"browser.test.assertRejects(browser.permissions.remove({'permissions': ['scripting']}))",
+
+        // getALL() should return all named permissions and granted match patterns.
+        @"let permissions = {'origins': ['*://webkit.org/*'], 'permissions': ['alarms', 'activeTab']}",
+        @"const result = await browser.permissions.getAll()",
+        @"browser.test.assertDeepEq(result, permissions)",
+
+        // Finish
+        @"browser.test.notifyPass()"
+    ]);
+
+    auto extension = adoptNS([[_WKWebExtension alloc] _initWithManifestDictionary:manifest resources:@{ @"background.js": backgroundScript }]);
+
+    auto manager = adoptNS([[TestWebExtensionManager alloc] initForExtension:extension.get()]);
+
+    // Grant "permissions" in the manifest.
+    for (NSString *permission in extension.get().requestedPermissions)
+        [manager.get().context setPermissionStatus:_WKWebExtensionContextPermissionStatusGrantedExplicitly forPermission:permission];
+
+    // Grant extension access to webkit.org.
+    _WKWebExtensionMatchPattern *matchPattern = [_WKWebExtensionMatchPattern matchPatternWithString:@"*://webkit.org/*"];
+    [manager.get().context setPermissionStatus:_WKWebExtensionContextPermissionStatusGrantedExplicitly forMatchPattern:matchPattern];
+
+    auto configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    configuration.get()._webExtensionController = manager.get().controller;
+
+    [manager loadAndRun];
+}
+
+TEST(WKWebExtensionAPIPermissions, AcceptPermissionsRequestTest)
+{
+    auto *manifest = @{
+        @"manifest_version": @3,
+        @"permissions": @[ @"alarms", @"activeTab" ],
+        @"optional_permissions": @[ @"webNavigation", @"cookies", @"declarativeNetRequest", @"*://*.apple.com/*" ],
```

Comment:
I think this is the problem: `@"*://*.apple.com/*"` should be in `optional_host_permission` since this is a v3 extension. It will be ignored in `optional_permissions`.